src/back/utils/validations.ts

```diff
@@ -102,3 +106,19 @@ export function validateDebugAddress(user: string | undefined) {
     throw new RequestError('Invalid user', RequestError.Unauthorized)
   }
 }
+
+export function validateDiscourseWebhookSignature(req: Request) {
+  const providedSignature = req.get('X-Discourse-Event-Signature') || ''
+  if (DISCOURSE_WEBHOOK_SECRET) {
+    const payload = req.body
+    const calculatedSignature = 'sha256='.concat(
+      crypto.createHmac('sha256', DISCOURSE_WEBHOOK_SECRET).update(JSON.stringify(payload)).digest('hex')
+    )
+
+    if (providedSignature !== calculatedSignature) {
+      throw new RequestError('Invalid signature', RequestError.Forbidden)
+    }
+  } else {
+    logger.warning('Unable to find secret for discourse webhook validation')
```

Comment:
I think we should also throw an error here so it doesn't keep the execution going wherever this is used 🤔